### PR TITLE
CupertinoTextField should not accept requestFocus when disabled

### LINKED ISSUE #65191: CupertinoTextField ignores `enabled` flag on focus request
A `CupertinoTextField` with a focus node and an `enabled` flag set to `false` becomes active when focus is requested for the node.

Here’s a snippet to reproduce the issue:
```
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(SampleWidget());
}

class SampleWidget extends StatelessWidget {
  final _focusNode = FocusNode();

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Center(
          child: Column(
            mainAxisSize: MainAxisSize.min,
            children: [
              CupertinoTextField(
                placeholder: "This field must be disabled",
                enabled: false,
                focusNode: _focusNode,
              ),
              RaisedButton(
                onPressed: () => FocusScope.of(context).requestFocus(_focusNode),
                child: Text("Request Focus"),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
```
I expect that a disabled field must stay disabled at all times. The only way to re-enable the field must be by explicitly setting its `enabled` flag to `true`.

This buggy behaviour seems to be specific to the Cupertino version of the text field. I haven’t experienced the same issue with the Material’s `TextField` widget.

<details>
  <summary>flutter doctor -v</summary>
<pre>
[✓] Flutter (Channel stable, 1.20.2, on Mac OS X 10.15.6 19G2021, locale en-UK)
    • Flutter version 1.20.2 at /Users/denis/Projects/src/flutter
    • Framework revision bbfbf1770c (3 weeks ago), 2020-08-13 08:33:09 -0700
    • Engine revision 9d5b21729f
    • Dart version 2.9.1

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/denis/Library/Android/sdk
    • Platform android-30, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.7)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.7, Build version 11E801a
    • CocoaPods version 1.10.0.beta.2

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] IntelliJ IDEA Ultimate Edition (version 2020.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA.app
    • Flutter plugin version 48.1.4
    • Dart plugin version 202.7206

[✓] Connected device (2 available)
    • Android SDK built for x86 (mobile) • emulator-5554                        • android-x86 • Android 10 (API 29) (emulator)
    • iPhone 11 Pro Max (mobile)         • B1F8BAAD-D347-487F-8227-D6AB6C78C2F6 • ios         • com.apple.CoreSimulator.SimRuntime.iOS-13-7 (simulator)

• No issues found!
</pre>
</details>

### PR DESCRIPTION
**Pull request by xu-baolin**
## Description

CupertinoTextField should not accept requestFocus when disabled

## Related Issues

Fixes #65191 

## Tests

I added the following tests:

See files.

### CODE REVIEW

File: packages/flutter/test/cupertino/text_field_test.dart [hunk 1]
```diff
@@ -2710,6 +2710,43 @@ void main() {
...
+        home: Center(
+          child: CupertinoTextField(
+            enabled: false,
+            focusNode: focusNode,
+          ),
+        ),
+      ),
+    );
+
+    // Should not accept requestFocus
```
Comment (pedromassango) [inline packages/flutter/test/cupertino/text_field_test.dart:2744]: Missing period here.

Comment (xu-baolin) [reply packages/flutter/test/cupertino/text_field_test.dart:2744]: > Missing period here.

Thanks for your review :)

You mean here need to call `await tester.pump()`, right?
What is the reason? Animation?

**Comment by xu-baolin** [comment]
@chunhtai Please review at your convenience, thanks.

**Comment by xu-baolin** [comment]
@HansMuller Ready for review :)

File: packages/flutter/test/cupertino/text_field_test.dart (hunk 1 above)
Comment (xu-baolin) [reply packages/flutter/test/cupertino/text_field_test.dart]: Fixed. Thanks.

**Comment by xu-baolin** [comment]
@justinmc @LongCatIsLooong please review, thanks.

**Review by LongCatIsLooong — APPROVED** [review]

**Comment by LongCatIsLooong** [comment]
Google testing failure doesn't seem related. Rerunning.

**Review by justinmc — APPROVED** [review]
LGTM and seems to be passing tests.  I'll add the "ready to merge" flag.

**Pull request merged**